Switch to `logging` and add `rich` text support

RBerga06 opened the pull request:
Risolve #2 e #3 (una volta che `logging` è integrato, non è troppo difficile aggiungere il supporto per `rich`)

RBerga06 committed:
Merge pull request #1 from RBerga06/develop

RBerga06 committed:
Merge branch 'feature-logging' of https://github.com/RBerga06/infn-lnl-stage2022-temaE into feature-logging

RBerga06 committed:
Update pyproject.toml

RBerga06 committed:
Update pyproject.toml

RBerga06 committed:
Prima bozza

pep8speaks commented:
Hello @RBerga06! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:







* In the file [`src/pi.py`](https://github.com/RBerga06/infn-lnl-stage2022-temaE/blob/4cbc296e065addf86cafa1ffd41efd1b7b3fe717/src/pi.py):

> [Line 52:9](https://github.com/RBerga06/infn-lnl-stage2022-temaE/blob/4cbc296e065addf86cafa1ffd41efd1b7b3fe717/src/pi.py#L52): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare 'except'
> [Line 79:9](https://github.com/RBerga06/infn-lnl-stage2022-temaE/blob/4cbc296e065addf86cafa1ffd41efd1b7b3fe717/src/pi.py#L79): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare 'except'











##### Comment last updated at 2022-07-12 15:36:20 UTC

RBerga06 committed:
" " -> " "

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Setup automatico: per attivarlo, basta importare i simboli da lì e non da `logger`

RBerga06 committed:
Update root.py

RBerga06 committed:
Aggiunto il supporto per -O

`python -O file.py` viene trattato più o meno come `python file.py -q`
(o, alternativamente, `python file.py` viene trattato più o meno come `python -O file.py -v`)

RBerga06 committed:
Migliorati alcuni commenti

RBerga06 committed:
Aggiunto un supporto maggiore per la parte destra

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Aggiunto il supporto per le `task`s

RBerga06 committed:
Fixed parens

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Esportati alcuni simboli

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Update root.py

RBerga06 committed:
Update spettro.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Risolto un problema con `_indent` e aggiornato `__all__`

RBerga06 committed:
Update pi.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Aggiunta la variabile d'ambiente NO_RICH

RBerga06 committed:
Aggiunto __slots__ a Logger

RBerga06 committed:
Aggiunte altre opzioni per `task`

RBerga06 committed:
Update root.py

RBerga06 committed:
Update root.py

RBerga06 committed:
Migliorati il colore degli errori e il prompt delle task

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Update root.py

RBerga06 committed:
Update .pep8speaks.yml

RBerga06 committed:
Update pi.py

RBerga06 committed:
Update pi.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Aggiunte due funzioni per scrivere a console

RBerga06 committed:
Semplificato il codice che scrive il titolo

RBerga06 committed:
Silenzia matplotlib

RBerga06 committed:
Update root.py

RBerga06 committed:
Update root.py

RBerga06 committed:
`task` può essere usato anche come decoratore

RBerga06 committed:
Aggiunta una funzione per ottenere il logger della task attuale

RBerga06 committed:
Update spettro.py

RBerga06 committed:
Update spettro.py

RBerga06 committed:
Update spettro.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Update pyproject.toml

RBerga06 committed:
Aggiunto il parametro `id` a `task()`

RBerga06 committed:
Update log.py

RBerga06 committed:
Update log.py

RBerga06 committed:
Update pyproject.toml

RBerga06 committed:
Update log.py

RBerga06 committed:
Migliorato un commento

RBerga06 committed:
Semplificata la definizione di TASK_MESSAGE

RBerga06 committed:
Update log.py

RBerga06 committed:
Aggiunte le `task` a `rand.py`

RBerga06 committed:
Update root.py

RBerga06 committed:
π colorato

RBerga06 committed:
Update log.py

RBerga06 committed:
Update pi.py

RBerga06 committed:
Update pi.py

RBerga06 committed:
Unicode box

RBerga06 committed:
Update spettro.py

RBerga06 committed:
Update classe.py

RBerga06 committed:
Update classe.py

RBerga06 committed:
Update ideal.py

RBerga06 committed:
Update stagisti.py

RBerga06 committed:
Rimossi i ` ` (non-breaking spaces)

RBerga06 committed:
PEP8 fixes

RBerga06 committed:
More PEP8 fixes

RBerga06 committed:
Update classe.py

RBerga06 reviewed (commented):
Tutto sembra a posto
